lightning/src/onion_message/messenger.rs

```diff
@@ -977,19 +1007,30 @@ where
 		)
 	}
 
-	fn handle_onion_message_response<T: OnionMessageContents>(
+	/// Handles the response to an [`OnionMessage`] based on its [`ResponseInstruction`],
+	/// enqueueing any response for sending.
+	pub fn handle_onion_message_response<T: OnionMessageContents> (
 		&self, response: ResponseInstruction<T>
-	) {
-		if let ResponseInstruction::WithoutReplyPath(response) = response {
-			let message_type = response.message.msg_type();
-			let _ = self.find_path_and_enqueue_onion_message(
-				response.message, Destination::BlindedPath(response.reply_path), None,
-				format_args!(
-					"when responding with {} to an onion message with path_id {:02x?}",
-					message_type,
-					response.path_id
-				)
-			);
+	) -> Result<Option<SendSuccess>, SendError> {
+		let (response, reply_path) = match response {
+			ResponseInstruction::WithReplyPath(response) => (response, self.create_blinded_path().ok()),
```

Comment:
Should we return an error if `create_blinded_path` fails? Otherwise, we'd send an onion message without a reply path.

Reply:
That sounds like a good idea! Updated in [2d769d6](https://github.com/lightningdevkit/rust-lightning/commit/2d769d683b2ae9de8d04c23c6bf24684b902eac4)
Thanks for the suggestion!